src/modules/fancyzones/lib/WindowMoveHandler.cpp

```diff
@@ -366,3 +380,26 @@ void WindowMoveHandlerPrivate::UpdateDragState(HWND window) noexcept
         }
     }
 }
+
+void WindowMoveHandlerPrivate::RestoreSize(HWND window) noexcept
+{
+    auto windowSizeData = GetPropW(window, RESTORE_SIZE_STAMP);
+    if (windowSizeData)
+    {
+        std::pair<int, int> windowSize;
+        memcpy(&windowSize, &windowSizeData, sizeof windowSize);

```

Comment:
that's UB, layout of `windowSize` could be changed.

Reply:
Are you suggesting to use our own custom struct?

Reply:
no, just assign to `.first`/`.second` or use a corresponding pair ctor

Reply:
PTAL, I used std::array, that should be safe

Reply:
alright, almost there, let's also use `windowSize.data()`